### PR TITLE
mpsutils: create a new language for check-in handlers

### PR DESCRIPTION
**Pull request by alexanderpann**
Create a new extension for CheckinExtensionPoint to add a new handler. You can find some example handlers in `com.mbeddr.mpsutil.checkinHandler.demo.plugin`. The handlers can accept a commit, reject it or execute arbitrary actions.

### CODE REVIEW

**Comment by sergej-koscejev** [comment]
I don't like the name `checkingHandler` because it looks to me like a misspelling of "check-in handler". Also, is it pre-commit handler or check-in handler? Let's use one term consistently as much as possible.

**Comment by alexanderpann** [comment]
@sergej-koscejev That was not intentional, I changed the name to `com.mbeddr.mpsutil.checkinHandler`.

File: code/languages/com.mbeddr.build/solutions/com.mbeddr.platform/models/com/mbeddr/platform/build.mps [hunk 1]
```diff
@@ -247,6 +247,7 @@
   <node concept="1l3spW" id="3AVJcIMlF8l">
     <property role="TrG5h" value="com.mbeddr.platform" />
     <property role="2DA0ip" value="../../../../../build/com.mbeddr.platform" />
+    <uo k="s:LocalSourcePathArtifact" v="i:23468bc7" />
```
Comment (sergej-koscejev) [inline code/languages/com.mbeddr.build/solutions/com.mbeddr.platform/models/com/mbeddr/platform/build.mps:250]: Looks like some user objects were stored in the model? Where do they come from? Seems wrong.

**Comment by alexanderpann** [comment]
Nice catch, I have fixed the build script but I don't know what happened.

File: code/languages/com.mbeddr.build/solutions/com.mbeddr.platform/models/com/mbeddr/platform/build.mps (hunk 1 above)
Comment (alexanderpann) [reply code/languages/com.mbeddr.build/solutions/com.mbeddr.platform/models/com/mbeddr/platform/build.mps]: Nice catch, I have fixed the build script but I don't know what happened.

**Comment by sergej-koscejev** [comment]
The plugin should ideally only register a single handler on the IDEA side and internally this handler should walk through extension point objects every time it gets called. I'm worried that the current implementation will cause "module classloader is disposed" errors during development when modules with the check-in handlers are recompiled and reloaded.

But I'm going to approve this PR as it is a step forward.

**Review by sergej-koscejev — APPROVED** [review]

**Comment by alexanderpann** [comment]
> module classloader is disposed

Yes, this is an issue, but shouldn't JetBrains try to find a fix? We have the same issue in other languages (e.g. iets3.opensource unit language) as you know.

**Comment by sergej-koscejev** [comment]
This is not something they can fix, it’s basically a memory leak. Alternatively there should be a listener that would update the list of extension objects every time a module is loaded or unloaded.

**Pull request merged**